app/client/src/utils/hooks/cyanWaterbodies.ts

```diff
@@ -0,0 +1,330 @@
+
+    if (!hucBoundaries?.features?.length) {
+      setHucData([]);
+      fetchedDataDispatch({
+        type: 'success',
+        id: localFetchedDataKey,
+        payload: [],
+      });
+      return;
+    }
+
+    if (services.status !== 'success') return;
+
+    const fetchPromise = fetchCyanWaterbodies(
+      hucBoundaries.features[0].geometry,
+      services.data,
+      controller.signal,
+    );
+
+    fetchAndTransformData(
+      fetchPromise,
+      fetchedDataDispatch,
+      'cyanWaterbodies',
+    ).then((data) => setHucData(data));
+
+    return function cleanup() {
+      controller.abort();
+    };
+  }, [fetchedDataDispatch, hucBoundaries, services]);
+
+  const extent = useRef<__esri.Polygon | null>(null);
+
+  const updateSurroundingData = useCallback(
+    async (abortSignal: AbortSignal) => {
+      if (services.status !== 'success') return;
+
+      if (!mapView) return;
+
+      // Same extent, no update necessary
+      if (mapView.extent === extent.current) return;
+      extent.current = mapView.extent;
+
+      await fetchAndTransformData(
+        fetchCyanWaterbodies(
+          Polygon.fromExtent(mapView.extent),
+          services.data,
+          abortSignal,
+        ),
+        fetchedDataDispatch,
+        surroundingFetchedDataKey,
+        hucData, // Filter out HUC data
+      );
+    },
+    [fetchedDataDispatch, hucData, mapView, services],
+  );
+
+  return updateSurroundingData;
+}
+
+/*
+## Utils
+*/
+
+function buildFeatures(data: CyanWaterbodyAttributes[]) {
+  return data.map((datum) => {
+    return new Graphic({
+      attributes: datum,
+      geometry: datum.geometry,
+    });
+  });
+}
+
+function buildLayer(
+  navigate: NavigateFunction,
+  services: ServicesState,
+  type: SublayerType,
+  getMapView: () => __esri.MapView,
+) {
+  const cyanWaterbodies = new FeatureLayer({
+    id:
+      type === 'enclosed'
+        ? `${localFetchedDataKey}`
+        : `${surroundingFetchedDataKey}`,
+    fields: [
+      { name: 'AREASQKM', type: 'double' },
+      { name: 'FID', type: 'integer' },
+      { name: 'GNIS_NAME', type: 'string' },
+      { name: 'locationName', type: 'string' },
+      { name: 'monitoringType', type: 'string', defaultValue: 'CyAN' },
+      { name: 'OBJECTID', type: 'oid' },
+      { name: 'oid', type: 'integer' },
+      {
+        name: 'orgName',
+        type: 'string',
+        defaultValue: 'Cyanobacteria Assessment Network (CyAN)',
+      },
+    ],
+    legendEnabled: false,
+    objectIdField: 'OBJECTID',
+    outFields: ['*'],
+    spatialReference: {
+      wkid: 102100,
+    },
+    popupTemplate: {
+      title: (feature: __esri.Feature) =>
+        getPopupTitle(feature.graphic.attributes),
+      content: (feature: __esri.Feature) =>
+        getPopupContent({
+          feature: feature.graphic,
+          mapView: getMapView(),
+          navigate,
+          services,
+        }),
+      outFields: ['*'],
+    },
```

Comment:
```suggestion
    popupTemplate: {
      title: getTitle,
      content: getTemplate,
      outFields: ['*'],
    },
```

This is related to getting the Change to this location section in the surrounding features popups.